File: Backend/internal/model/event.go
package model

import "time"

// Event ...
type Event struct {
	ID          int       `json:"eventID"`
	BeginTime   time.Time `json:"beginTime"`
	EndTime     time.Time `json:"endTime"`
	Description string    `json:"description,omitempty"`
	MeetingLink string    `json:"meetingLink"`
	IsRepeating bool      `json:"isRepeating"`
	Title       string    `json:"title"`
	Owner       string    `json:"owner"`
}

File: Frontend/src/types/event/event.ts
export interface EventState {
  eventID: number;
  title: string;
  beginTime: string;
  endTime: string;
  description: string;
  meetingLink: string;
  isRepeating: boolean;
  owner: string;
  notificationTime: number;
  notificationInTelegram: boolean;
  notificationInEmail: boolean;
  notificationInSMS: boolean;
  participiants: number[];
}

export enum EventActionTypes {
  SAVE_EVENT = "SAVE_EVENT",
}

interface SaveEventAction {
  type: EventActionTypes.SAVE_EVENT;
  payload: EventState;
}

export type EventAction = SaveEventAction;

export interface EventCreationState {
  title: string;
  beginTime: string;
  endTime: string;
  description: string;
  meetingLink: string;
  isRepeating: boolean;
  notificationTime: number;
  notificationInTelegram: boolean;
  notificationInEmail: boolean;
  notificationInSMS: boolean;
  participiants: number[];
}
